### PR TITLE
loki: define querier startup probe

### PR DESCRIPTION
**Pull request by sreis**
A workaround to ensure the queriers are initialized properly.

From https://github.com/grafana/loki/issues/3085. The queriers initialize the storage credentials in the query evaluation code path. To ensure the querier is initialized before being marked ready to receive requests we query the labels endpoint in the system tenant until we get a 200 OK. The system tenant is created by default. This means the queriers startupProbe waits for the ingesters to receive some data.

The startupProbe indicates whether the application within the container is started. All other probes are disabled if a startup probe is provided, until it succeeds. The readinessProbe indicates whether the container is ready to respond to requests. If the readiness probe fails, the endpoints controller removes the Pod's IP address from the endpoints of all Services that match the Pod.

https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/
https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-startup-probes

### CODE REVIEW

File: packages/controller/src/resources/loki/index.ts [hunk 1]
```diff
@@ -1289,7 +1289,47 @@ export function LokiResources(
...
+                      port: 1080 as any,
+                      httpHeaders: [
+                        {
+                          name: "X-Scope-OrgID",
+                          value: "system"
+                        }
+                      ]
+                    },
+                    // We are being liberal with these settings to cover a worst
+                    // case startup time.
```
Comment (jgehrcke) [inline packages/controller/src/resources/loki/index.ts:1326]: :muscle:

Comment (sreis) [reply packages/controller/src/resources/loki/index.ts:1326]: :crossed_fingers:

**Comment by sreis** [comment]
Running the CI job for upgrades and seems to be still hitting #905. Creating a local cluster for additional testing and I'll report back.

**Comment by sreis** [comment]
Looking at the [CI logs](https://buildkite.com/organizations/opstrace/pipelines/prs/builds/4782/jobs/7422f6e3-bcc5-4968-965e-c4cfa86b655f/raw_log) for this PR we see that hitting the labels endpoint, when the cluster creation finishes, no longer hit any timeout issues. This means the Loki queries were initialized properly.

```
[2021-06-21T20:34:54Z] 2021-06-21T20:34:54.870Z [32minfo[39m: waiting for expected HTTP responses at these URLs: {
[2021-06-21T20:34:54Z]   "https://cortex.default.prs-bk-4782-bc4-a.opstrace.io/api/v1/labels": "default",
[2021-06-21T20:34:54Z]   "https://loki.default.prs-bk-4782-bc4-a.opstrace.io/loki/api/v1/labels": "default",
[2021-06-21T20:34:54Z]   "https://cortex.system.prs-bk-4782-bc4-a.opstrace.io/api/v1/labels": "system",
[2021-06-21T20:34:54Z]   "https://loki.system.prs-bk-4782-bc4-a.opstrace.io/loki/api/v1/labels": "system"
[2021-06-21T20:34:54Z] }
[2021-06-21T20:34:54Z] 2021-06-21T20:34:54.998Z [32minfo[39m: https://cortex.default.prs-bk-4782-bc4-a.opstrace.io/api/v1/labels: got expected HTTP response
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.002Z [32minfo[39m: https://loki.default.prs-bk-4782-bc4-a.opstrace.io/loki/api/v1/labels: got expected HTTP response
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.078Z [32minfo[39m: https://loki.system.prs-bk-4782-bc4-a.opstrace.io/loki/api/v1/labels: got expected HTTP response
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.093Z [32minfo[39m: https://cortex.system.prs-bk-4782-bc4-a.opstrace.io/api/v1/labels: got expected HTTP response
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.094Z [32minfo[39m: wait for data API endpoints: all probe URLs returned expected HTTP responses, continue
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.094Z [32minfo[39m: waiting for expected HTTP responses at these URLs: {
[2021-06-21T20:34:55Z]   "https://dd.default.prs-bk-4782-bc4-a.opstrace.io/api/v1/series": "default"
[2021-06-21T20:34:55Z] }
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.180Z [32minfo[39m: wait for DD API endpoints: all probe URLs returned expected HTTP responses, continue
[2021-06-21T20:34:55Z] 2021-06-21T20:34:55.180Z [32minfo[39m: waiting for expected HTTP responses at these URLs: {
[2021-06-21T20:34:55Z]   "https://default.prs-bk-4782-bc4-a.opstrace.io/": "default",
[2021-06-21T20:34:55Z]   "https://system.prs-bk-4782-bc4-a.opstrace.io/": "system"
[2021-06-21T20:34:55Z] }
```

**Comment by sreis** [comment]
Rebased with main to trigger more tests.

**Comment by jgehrcke** [comment]
@sreis would we want to go ahead with this? :) Maybe do another CI run after rebasing on latest main?

**Comment by sreis** [comment]
I'll wait for #967 to be closed before rebasing again.

**Comment by sreis** [comment]
GCP run failed browser tests with known flake #920 .

**Pull request merged**